secret-sync-controller/main.go

```diff
@@ -0,0 +1,156 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"os"
+	"time"
+
+	// Import all Kubernetes client auth plugins (e.g. Azure, GCP, OIDC, etc.)
+	// to ensure that exec-entrypoint and run can make use of them.
+	"google.golang.org/grpc"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/metrics/server"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	secretsstorecsiv1 "sigs.k8s.io/secrets-store-csi-driver/apis/v1"
+	secretsyncv1alpha1 "sigs.k8s.io/secrets-store-csi-driver/secret-sync-controller/api/v1alpha1"
+	"sigs.k8s.io/secrets-store-csi-driver/secret-sync-controller/controllers"
+	"sigs.k8s.io/secrets-store-csi-driver/secret-sync-controller/pkg/k8s"
+	"sigs.k8s.io/secrets-store-csi-driver/secret-sync-controller/pkg/provider"
+	"sigs.k8s.io/secrets-store-csi-driver/secret-sync-controller/pkg/util/tokenutil"
+	//+kubebuilder:scaffold:imports
+)
+
+var (
+	scheme   = runtime.NewScheme()
+	setupLog = ctrl.Log.WithName("setup")
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+
+	utilruntime.Must(secretsyncv1alpha1.AddToScheme(scheme))
+
+	utilruntime.Must(secretsstorecsiv1.AddToScheme(scheme))
+	//+kubebuilder:scaffold:scheme
+}
+
+func main() {
+	var metricsAddr string
+	var enableLeaderElection bool
+	var probeAddr string
+	var tokenRequestAudience string
+	var providerVolumePath string
+	var maxCallRecvMsgSize int
+	var rotationPollIntervalInSeconds int64
+
+	flag.Int64Var(&rotationPollIntervalInSeconds, "rotation-poll-interval-in-seconds", 120, "Poll interval in seconds for token rotation.")
+	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8085", "The address the metric endpoint binds to.")
+	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
+		"Enable leader election for controller manager. "+
+			"Enabling this will ensure there is only one active controller manager.")
+	flag.StringVar(&providerVolumePath, "provider-volume", "/provider", "Volume path for provider.")
+	flag.IntVar(&maxCallRecvMsgSize, "max-call-recv-msg-size", 1024*1024*4, "maximum size in bytes of gRPC response from plugins")
+	flag.StringVar(&tokenRequestAudience, "token-request-audience", "api://AzureADTokenExchange", "Audience for the token request.")
```

Comment:
This field should not have a default. 

If no audience is specified, do we want to skip generating a token for the service account token?